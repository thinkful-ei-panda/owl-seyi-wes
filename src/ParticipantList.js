import React from 'react';
import Participant from './Participant';

function ParticipantList(props){
    return (
        <section className='participant-list'>
        {
        props.participants.map((arrEle,index) => <Participant key={props.participants[index].id} participant={props.participants[index]} />)
        }
        </section>
        
    )
}

export default ParticipantList;