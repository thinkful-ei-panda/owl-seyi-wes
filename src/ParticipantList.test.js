import React from 'react';
import ReactDOM from 'react-dom';
import ParticipantList from './ParticipantList';
import store from './store';

import renderer from 'react-test-renderer';


describe('Participant List', () => {
  it('renders without crashing', () => {

    const div = document.createElement('div');
    ReactDOM.render(<ParticipantList participants={[...store.participants]} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<ParticipantList participants={[...store.participants]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

});

