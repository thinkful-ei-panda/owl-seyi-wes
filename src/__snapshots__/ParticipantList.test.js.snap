// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participant List renders the UI as expected 1`] = `
<section
  className="participant-list"
>
  <div
    className="participant-container"
  >
    <div
      className="left small"
    >
      <img
        alt="avatar"
        src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
      />
    </div>
    <div
      className="right"
    >
      <div>
        Koren Templeton
      </div>
      <div>
        on stage
      </div>
    </div>
  </div>
  <div
    className="participant-container"
  >
    <div
      className="left small"
    >
      <img
        alt="avatar"
        src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
      />
    </div>
    <div
      className="right"
    >
      <div>
        Caty Flucker
      </div>
      <div>
        in session
      </div>
    </div>
  </div>
  <div
    className="participant-container"
  >
    <div
      className="left small"
    >
      <img
        alt="avatar"
        src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
      />
    </div>
    <div
      className="right"
    >
      <div>
        Frank Runciman
      </div>
      <div>
        in session
      </div>
    </div>
  </div>
  <div
    className="participant-container"
  >
    <div
      className="left small"
    >
      <img
        alt="avatar"
        src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
      />
    </div>
    <div
      className="right"
    >
      <div>
        Ashla Attwool
      </div>
      <div>
        on stage
      </div>
    </div>
  </div>
</section>
`;
